camel-k --runtime-version is not honoured for all runtimes

I've tried to use the 0.3.1 runtime version from a staging repository before it was published on maven central but the camel-k-operator issued the log below:
`[ERROR] Plugin org.apache.camel.k:camel-k-maven-plugin:0.3.1 or one of its dependencies could not be resolved: Failure to find org.apache.camel.k:camel-k-maven-plugin:jar:0.3.1 in https://repo.maven.apache.org/maven2`

According to this [discussion](https://www.mail-archive.com/dev@camel.apache.org/msg44720.html), chances are that the cause would be a missing PluginRepositories near [there](https://github.com/apache/camel-k/blob/master/pkg/builder/springboot/generator.go#L42).

--- a/pkg/builder/builder_steps.go
+++ b/pkg/builder/builder_steps.go
@@ -48,23 +48,6 @@ func GenerateProject(ctx *Context) error {
 	ctx.Project = p
 
 	//
-	// Repositories
-	//
-
-	ctx.Project.Repositories = make([]maven.Repository, 0, len(ctx.Request.Repositories))
-	ctx.Project.PluginRepositories = make([]maven.Repository, 0, len(ctx.Request.Repositories))
-
-	for i, r := range ctx.Request.Repositories {
-		repo := maven.NewRepository(r)
-		if repo.ID == "" {
-			repo.ID = fmt.Sprintf("repo-%03d", i)
-		}
-
-		ctx.Project.Repositories = append(ctx.Project.Repositories, repo)
-		ctx.Project.PluginRepositories = append(ctx.Project.PluginRepositories, repo)
-	}
-
-	//
 	// set-up dependencies
 	//
 

--- a/pkg/builder/builder_steps_test.go
+++ b/pkg/builder/builder_steps_test.go
@@ -28,6 +28,67 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestMavenRepositories(t *testing.T) {
+	catalog, err := test.DefaultCatalog()
+	assert.Nil(t, err)
+
+	ctx := Context{
+		Catalog: catalog,
+		Request: Request{
+			Catalog:        catalog,
+			RuntimeVersion: defaults.RuntimeVersion,
+			Platform: v1alpha1.IntegrationPlatformSpec{
+				Build: v1alpha1.IntegrationPlatformBuildSpec{
+					CamelVersion: catalog.Version,
+				},
+			},
+			Dependencies: []string{
+				"runtime:jvm",
+			},
+			Repositories: []string{
+				"https://repository.apache.org/content/repositories/snapshots@id=apache.snapshots@snapshots@noreleases",
+				"https://oss.sonatype.org/content/repositories/snapshots/@id=sonatype.snapshots@snapshots",
+			},
+		},
+	}
+
+	err = GenerateProject(&ctx)
+	assert.Nil(t, err)
+	err = InjectDependencies(&ctx)
+	assert.Nil(t, err)
+
+	assert.Equal(t, 2, len(ctx.Project.Repositories))
+	assert.Equal(t, 2, len(ctx.Project.PluginRepositories))
+
+	assert.Contains(t, ctx.Project.Repositories, maven.Repository{
+		ID:        "apache.snapshots",
+		URL:       "https://repository.apache.org/content/repositories/snapshots",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: false},
+	})
+
+	assert.Contains(t, ctx.Project.Repositories, maven.Repository{
+		ID:        "sonatype.snapshots",
+		URL:       "https://oss.sonatype.org/content/repositories/snapshots/",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: true},
+	})
+
+	assert.Contains(t, ctx.Project.PluginRepositories, maven.Repository{
+		ID:        "apache.snapshots",
+		URL:       "https://repository.apache.org/content/repositories/snapshots",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: false},
+	})
+
+	assert.Contains(t, ctx.Project.PluginRepositories, maven.Repository{
+		ID:        "sonatype.snapshots",
+		URL:       "https://oss.sonatype.org/content/repositories/snapshots/",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: true},
+	})
+}
+
 func TestGenerateJvmProject(t *testing.T) {
 	catalog, err := test.DefaultCatalog()
 	assert.Nil(t, err)

--- a/pkg/builder/builder_utils.go
+++ b/pkg/builder/builder_utils.go
@@ -19,6 +19,7 @@ package builder
 
 import (
 	"encoding/xml"
+	"fmt"
 
 	"github.com/apache/camel-k/pkg/util/defaults"
 
@@ -40,6 +41,9 @@ func ArtifactIDs(artifacts []v1alpha1.Artifact) []string {
 
 // NewProject --
 func NewProject(ctx *Context) (maven.Project, error) {
+	//
+	// Catalog
+	//
 	if ctx.Catalog == nil {
 		c, err := camel.Catalog(ctx.Request.C, ctx.Client, ctx.Namespace, ctx.Request.Platform.Build.CamelVersion)
 		if err != nil {
@@ -71,6 +75,23 @@ func NewProject(ctx *Context) (maven.Project, error) {
 			},
 		},
 		Dependencies: make([]maven.Dependency, 0),
+	}
+
+	//
+	// Repositories
+	//
+
+	p.Repositories = make([]maven.Repository, 0, len(ctx.Request.Repositories))
+	p.PluginRepositories = make([]maven.Repository, 0, len(ctx.Request.Repositories))
+
+	for i, r := range ctx.Request.Repositories {
+		repo := maven.NewRepository(r)
+		if repo.ID == "" {
+			repo.ID = fmt.Sprintf("repo-%03d", i)
+		}
+
+		p.Repositories = append(p.Repositories, repo)
+		p.PluginRepositories = append(p.PluginRepositories, repo)
 	}
 
 	return p, nil

--- a/pkg/builder/springboot/generator.go
+++ b/pkg/builder/springboot/generator.go
@@ -35,21 +35,6 @@ func GenerateProject(ctx *builder.Context) error {
 	ctx.Project = p
 
 	//
-	// Repositories
-	//
-
-	ctx.Project.Repositories = make([]maven.Repository, 0, len(ctx.Request.Repositories))
-
-	for i, r := range ctx.Request.Repositories {
-		repo := maven.NewRepository(r)
-		if repo.ID == "" {
-			repo.ID = fmt.Sprintf("repo-%03d", i)
-		}
-
-		ctx.Project.Repositories = append(ctx.Project.Repositories, repo)
-	}
-
-	//
 	// set-up dependencies
 	//
 

--- a/pkg/builder/springboot/generator_test.go
+++ b/pkg/builder/springboot/generator_test.go
@@ -1,0 +1,90 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package springboot
+
+import (
+	"testing"
+
+	"github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
+	"github.com/apache/camel-k/pkg/builder"
+	"github.com/apache/camel-k/pkg/util/defaults"
+	"github.com/apache/camel-k/pkg/util/maven"
+	"github.com/apache/camel-k/pkg/util/test"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMavenRepositories(t *testing.T) {
+	catalog, err := test.DefaultCatalog()
+	assert.Nil(t, err)
+
+	ctx := builder.Context{
+		Catalog: catalog,
+		Request: builder.Request{
+			Catalog:        catalog,
+			RuntimeVersion: defaults.RuntimeVersion,
+			Platform: v1alpha1.IntegrationPlatformSpec{
+				Build: v1alpha1.IntegrationPlatformBuildSpec{
+					CamelVersion: catalog.Version,
+				},
+			},
+			Dependencies: []string{
+				"runtime:jvm",
+			},
+			Repositories: []string{
+				"https://repository.apache.org/content/repositories/snapshots@id=apache.snapshots@snapshots@noreleases",
+				"https://oss.sonatype.org/content/repositories/snapshots/@id=sonatype.snapshots@snapshots",
+			},
+		},
+	}
+
+	err = GenerateProject(&ctx)
+	assert.Nil(t, err)
+	err = builder.InjectDependencies(&ctx)
+	assert.Nil(t, err)
+
+	assert.Equal(t, 2, len(ctx.Project.Repositories))
+	assert.Equal(t, 2, len(ctx.Project.PluginRepositories))
+
+	assert.Contains(t, ctx.Project.Repositories, maven.Repository{
+		ID:        "apache.snapshots",
+		URL:       "https://repository.apache.org/content/repositories/snapshots",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: false},
+	})
+
+	assert.Contains(t, ctx.Project.Repositories, maven.Repository{
+		ID:        "sonatype.snapshots",
+		URL:       "https://oss.sonatype.org/content/repositories/snapshots/",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: true},
+	})
+
+	assert.Contains(t, ctx.Project.PluginRepositories, maven.Repository{
+		ID:        "apache.snapshots",
+		URL:       "https://repository.apache.org/content/repositories/snapshots",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: false},
+	})
+
+	assert.Contains(t, ctx.Project.PluginRepositories, maven.Repository{
+		ID:        "sonatype.snapshots",
+		URL:       "https://oss.sonatype.org/content/repositories/snapshots/",
+		Snapshots: maven.RepositoryPolicy{Enabled: true},
+		Releases:  maven.RepositoryPolicy{Enabled: true},
+	})
+}

--- a/pkg/controller/integrationplatform/initialize.go
+++ b/pkg/controller/integrationplatform/initialize.go
@@ -115,7 +115,7 @@ func (action *initializeAction) Handle(ctx context.Context, ip *v1alpha1.Integra
 			action.L.Info("Repositories:")
 		}
 
-		action.L.Infof("    %d - %s", i, r)
+		action.L.Infof("%d - %s", i, r)
 	}
 
 	err = action.client.Update(ctx, target)

